src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1190,6 +1190,31 @@ export class SavedObjectsRepository {
 
     const time = this._getCurrentTime();
 
+    let rawUpsert: SavedObjectsRawDoc | undefined;
+    if (upsert) {
+      let savedObjectNamespace: string | undefined;
+      let savedObjectNamespaces: string[] | undefined;
+
+      if (this._registry.isSingleNamespace(type) && namespace) {
+        savedObjectNamespace = namespace;
+      } else if (this._registry.isMultiNamespace(type)) {
+        savedObjectNamespaces = await this.preflightGetNamespaces(type, id, namespace);
+      }
+
+      const migrated = this._migrator.migrateDocument({
+        id,
+        type,
+        ...(savedObjectNamespace && { namespace: savedObjectNamespace }),
+        ...(savedObjectNamespaces && { namespaces: savedObjectNamespaces }),
+        attributes: {
+          ...upsert,
+          ...attributes,
+        },
```

Comment:
Should we invert the order here?

AFAIUnderstand from [the docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html#upserts), `upsert` works as: if the document does not exist, create the document with the values from `upsert` (and do not apply any of the changes un the _update_ part).

I understand that merging it with attributes is to ensure that all fields are descripted, but in case of clashing, I would expect the keys in upsert to persist in the upsert scenario. WDYT?

Reply:
> AFAIUnderstand from the docs, upsert works as: if the document does not exist, create the document with the values from upsert (and do not apply any of the changes un the update part).

Yea, that is right.

My thinking about this API was: 

when `upsert` is specified and the document doesn't exist:
- Use the upsert attribute as if you were performing a `create` operation with them
- Then perform an update with the `attributes` provided

With means having the `attributes` properties prevails over `upsert` properties in the upsert option we're sending to ES.

But I don't mind switching the order. We could even, to get closer to ES, just ignores the `attributes` in the `upsert` sent to ES.

I don't really have a strong opinion on that, as long as we document it.

Reply:
> I don't really have a strong opinion on that, as long as we document it.

I feel the same. If I had to pick one, I guess I'd prefer to emulate ES's behaviour since we are using a similar API, and I would expect devs to think that it works similarly. Alternatively, we could use a different name or API format for `upsert` to make it clear that it won't work the same.

Reply:
Done in https://github.com/elastic/kibana/pull/98712/commits/c4a2a5763f3b3c8a7f4f0c2e229fca091137b0e4